### PR TITLE
[RHELC-1288] Add integration test verifying duplicate pkgs installed

### PR DESCRIPTION
**Pull request by kokesak**
Verify that the conversion does not crash when the same package (of different version) is installed on the system.
Verify that the proper inhibitor is raised. The bug is reproducible only on EL7 distros. On EL8 the issue is handled by the DNF itself.

Address the https://issues.redhat.com/browse/RHELC-1070

Jira Issues: [RHELC-1070](https://issues.redhat.com/browse/RHELC-1070),  [RHELC-1288](https://issues.redhat.com/browse/RHELC-1288)

Checklist

- [ ] PR has been tested manually in a VM (either author or reviewer)
- [ ] Jira issue has been made public if possible
- [ ] `[RHELC-]` is part of the PR title 
- [ ] GitHub label has been added to help with Release notes 
- [ ] PR title explains the change from the user's point of view
- [ ] Code and tests are documented properly
- [ ] The commits are squashed to as few commits as possible (without losing data) 
- [ ] When merged: Jira issue has been updated to `Release Pending` if relevant

### CODE REVIEW

**Comment by kokesak** [comment]
CentOS7: http://artifacts.osci.redhat.com/testing-farm/bc4016f9-3385-4f85-8cb3-e9a3c817cced/

File: tests/integration/tier0/non-destructive/duplicate-pkgs/main.fmf [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    Verify that the proper inhibitor is raised.
+
+
+/duplicate-pkgs:
+    summary+: |
+        Verify the https://issues.redhat.com/browse/RHELC-1070. The bug is reproducible only
+        on EL7 distros. On EL8 the issue is handled by the DNF itself.
+    adjust+:
+        - enabled: false
+          when: distro == centos-8, alma-8, rocky-8, oracle-8
```
Comment (danmyway) [inline tests/integration/tier0/non-destructive/duplicate-pkgs/main.fmf:15]:
```suggestion
          when: distro == centos-8, alma-8, rocky-8, oracle-8
```
Please add the `because:` section.
(e.g. "Reproducible on/reported for 7.9 RHEL7-like distributions only". Or what you have in the summary The bug is reproducible only on EL7 distros. On EL8 the issue is handled by the DNF itself.)

File: tests/integration/tier0/non-destructive/duplicate-pkgs/main.fmf [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+description+: |
+    Verify that the conversion does not crash when the same
+    package (of different version) is installed on the system.
+    Verify that the proper inhibitor is raised.
+
+
+/duplicate-pkgs:
+    summary+: |
+        Verify the https://issues.redhat.com/browse/RHELC-1070. The bug is reproducible only
+        on EL7 distros. On EL8 the issue is handled by the DNF itself.
```
Comment (danmyway) [inline tests/integration/tier0/non-destructive/duplicate-pkgs/main.fmf:12]:
```suggestion
    link: https://issues.redhat.com/browse/RHELC-1070        
```
I think the summary can be removed from here.
It doesn't say much new and can be replaced by the link and because keys.
The summary should be ideally 50 characters, since there is just one test case, it can be inherited from above.

File: tests/integration/tier0/non-destructive/duplicate-pkgs/test_duplicate_pkgs.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    nodejs from the epel repository
+    """
+    with convert2rhel(
+        "analyze -y --serverurl {} --username {} --password {} --debug".format(
+            env.str("RHSM_SERVER_URL"),
+            env.str("RHSM_USERNAME"),
+            env.str("RHSM_PASSWORD"),
+        )
+    ) as c2r:
+        # The error about duplicate packages should be reported at the end inside the report section
```
Comment (danmyway) [inline tests/integration/tier0/non-destructive/duplicate-pkgs/test_duplicate_pkgs.py:65]:
```suggestion
        # The error about duplicate packages should be included at the end of the pre-conversion analysis report
```

File: tests/integration/tier0/non-destructive/duplicate-pkgs/test_duplicate_pkgs.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+@pytest.mark.test_duplicate_pkgs
+def test_list_third_party_pkgs_error(convert2rhel, install_duplicate_pkg):
+    """
+    This test verifies, that the  TASK - [Convert: List third-party packages]
+    won't fail listing packages if previously problematic third party packages are installed.
+    Installed package(s):
+    v8-devel from the epel repository
+    nodejs from the epel repository
+    """
```
Comment (danmyway) [inline tests/integration/tier0/non-destructive/duplicate-pkgs/test_duplicate_pkgs.py:57]:
```suggestion
def test_list_third_party_pkgs_error(convert2rhel, install_duplicate_pkg):
    """
    This test verifies, that the  TASK - [Convert: List third-party packages]
    won't fail listing packages if previously problematic third party packages are installed.
    Installed package(s):
    v8-devel from the epel repository
    nodejs from the epel repository
    """
    
```
This seems to be out of place.

File: tests/integration/tier0/non-destructive/duplicate-pkgs/test_duplicate_pkgs.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    shell(f"rpm -i --noscripts --justdb --nodeps --force duplicate-{pkg}.rpm")
+
+    assert int(shell(f"rpm -q {pkg} | wc -l ").output) >= 2
+
+    yield
+
+    # This should remove both the packages
+    shell(f"yum remove -y {pkg}")
+
+    # If the package (single one) was installed already on the system, then install it back
```
Comment (danmyway) [inline tests/integration/tier0/non-destructive/duplicate-pkgs/test_duplicate_pkgs.py:44]:
```suggestion
    # If the package was originally installed on the system, install it back
```

File: tests/integration/tier0/non-destructive/duplicate-pkgs/main.fmf (hunk 1 above)
Comment (kokesak) [reply tests/integration/tier0/non-destructive/duplicate-pkgs/main.fmf:13]: Right, I added the `because:` section

File: tests/integration/tier0/non-destructive/duplicate-pkgs/main.fmf (hunk 2 above)
Comment (kokesak) [reply tests/integration/tier0/non-destructive/duplicate-pkgs/main.fmf]: Good idea. Added a `link` there instead.

File: tests/integration/tier0/non-destructive/duplicate-pkgs/test_duplicate_pkgs.py (hunk 5 above)
Comment (kokesak) [reply tests/integration/tier0/non-destructive/duplicate-pkgs/test_duplicate_pkgs.py]: Done
